### PR TITLE
[RFC] New cli

### PR DESCRIPTION
**Pull request by tmarcu**
Use the cobra package to define a new CLI interface with more proper
structure, utilizing the client library package for operations.

Signed-off-by: Tudor Marcu <tudor.marcu@intel.com>

### CODE REVIEW

File: ciao/cmd/delete.go [hunk 1]
```diff
@@ -153,6 +153,28 @@ var volumeDelCmd = &cobra.Command{
 	},
 }
 
+func deleteTenant(args []string) int {
+	tenantID := args[0]
+
+	err := c.DeleteTenant(tenantID)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error deleting volume: %s\n", err)
```
Comment (tmarcu) [inline ciao/cmd/delete.go]: This should probably say tenant and not volume?

**Comment by rbradford** [comment]
@tmarcu i've done a squash as it was getting out of hand. I'll keep posting new commits until Wednesday when i'll squash again.

File: ciao/main.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/ciao-project/ciao/ciao/cmd"
+)
+
+func main() {
+	cmd.Execute()
```
Comment (markdryan) [inline ciao/main.go]: Help seems to be missing for some of the commands:

```
Available Commands:
  add         
  attach      
  create      Create cluster objects.
  delete      
  detach      Detach various ciao objects
  evacuate    
  help        Help about any command
  list        List information about various ciao objects
  remove      
  restart     
  restore     Restore a node
  show        Show detailed information about a ciao object
  stop        
  update      Update various ciao objects

```

File: ciao/cmd/create.go [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+		}
+		defer func() { _ = f.Close() }()
+
+		imageVisibility := types.Private
+		if imgFlags.visibility != "" {
+			imageVisibility = types.Visibility(imgFlags.visibility)
+			switch imageVisibility {
+			case types.Public, types.Private, types.Internal:
+			default:
+
```
Comment (markdryan) [inline ciao/cmd/create.go]: weird newline here

File: bat/compute.go [hunk 4]
```diff
@@ -103,15 +103,6 @@ type NodeStatus struct {
 	Hostname              string    `json:"hostname"`
 }
```
Comment (markdryan) [inline bat/compute.go]: Don't we need to delete RunCIAOCLI and friends or are they still being used somewhere?

File: ciao/cmd/remove.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+var removeExternalIPCmd = &cobra.Command{
+	Use:   "external-ip POOL SUBNET or IP",
+	Short: "Remove IP address or subnet from the pool",
+	Args:  cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		name := args[0]
+
+		_, network, err := net.ParseCIDR(args[1])
```
Comment (markdryan) [inline ciao/cmd/remove.go]: Is this args[1] safe here?

Comment (rbradford) [reply ciao/cmd/remove.go]: Yeh but it should be .ExactArgs(2) one for the pool name and one for the subnet.

File: ciao/cmd/create.go (hunk 3 above)
Comment (rbradford) [reply ciao/cmd/create.go]: Fixed.

File: ciao/cmd/create.go [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+		return nil, errors.New("Invalid workload yaml: no bootable disks specified for a VM")
+	}
+
+	return storage, nil
+}
+
+func optToReq(opt workloadOptions, req *types.Workload) error {
+	b, err := ioutil.ReadFile(opt.CloudConfigFile)
+	if err != nil {
+		return err
```
[the remark this answers is not in the document]
Comment (rbradford) [reply ciao/cmd/create.go]: I'm happy with the wrapping at the caller point for this one.

File: ciao/main.go (hunk 2 above)
Comment (rbradford) [reply ciao/main.go]: Fixed this by rolling out more Short: fields.

**Pull request merged**